app/helpers/Intersect.js

```diff
@@ -571,3 +572,41 @@ export function disableDebugMode() {
   // Kick off intersection calculations to restore data
   checkIntersect();
 }
+
+async function getPageData(authority, page) {
+  // this function should always return an object.
+  // if there's no data, return an empty object.
+  const cacheKey = authority.internal_id + '|page:' + page.id;
+  let pageData = await GetStoreData(cacheKey);
+  if (!pageData) {
+    if (await shouldDownloadPageData()) {
+      try {
+        pageData = await retrieveUrlAsJson(page.filename);
+        SetStoreData(cacheKey, pageData);
+      } catch (e) {
+        // sometimes the url isn't right, the download will fail
+        pageData = {};
+      }
+    } else {
+      // no cache and can't download, skip this page.
+      pageData = {};
```

Comment:
Could we invert the nested logic to have a simpler tree here? Something like:

```typescript
const emptyPageData = {}
let pageData = await GetStoreData(cacheKey);

if (!pageData && await shouldDownloadPageData()) {
  try {
        pageData = await retrieveUrlAsJson(page.filename);
        SetStoreData(cacheKey, pageData);
    } catch (e) {
       // Maybe we need to log this or notify somehow of the broken URL?
    }
   return pageData || emptyPageData;
}
```

Reply:
to clarify, the suggested code looks the same but just combines `if (!pageData) {` and `if (await shouldDownloadPageData()) {` into one statement right?

Regarding the comment about error thrown, I wasn't aware of any logging, are you saying just console logging?

Reply:
You're right! Is just a combination to remove one branch of the decision flow, not at all an improvement but if we feel that one less branch could be beneficial maybe worth doing. About the logging, I'm not aware of ay aggregator service we are using but if not retrieving page data is something that needs correction on a third party, some visibility of these errors could help with this goal. Not sure if just logging it in the device logs would suffice or if not having a page available would grant any action, but if it does it might be worth storing some log info whether here or on a logging service that I am not sure we have at the moment.